src/Compilers/Core/Portable/Symbols/IAssemblySymbol.cs

```diff
@@ -76,6 +77,11 @@ public interface IAssemblySymbol : ISymbol
         /// </summary>
         INamedTypeSymbol? ResolveForwardedType(string fullyQualifiedMetadataName);
 
+        /// <summary>
+        /// Returns type symbols for top-level (non-nested) types forwarded by this assembly.
```

Comment:
> How do nested types participate in type forwarding, if they participate at all?

Compilers emit metadata information describing forwarding of nested types. However, compilers themselves don't use this information because a nested type cannot be forwarded in a different place, it should still be inside the same containing type. So, they necessarily move together and if you locate the container, no need to go through a complicated lookup for a nested type. Also, compilers don't even allow using a nested type as an argument for TypeForwardedTo attribute. 
Bottom line:
- I believe the information about forwarded nested types is redundant.
- Including that information would require more changes within compilers and as a result would increase implementation and testing cost.

---
In reply to: [426731695](https://github.com/dotnet/roslyn/pull/44307#discussion_r426731695) [](ancestors = 426731695)